backend/main.cpp

```diff
@@ -0,0 +1,199 @@
+// *********************************************************************
+// ||                            INCLUDES                             ||
+// *********************************************************************
+#include <iostream>
+#include <boost/asio.hpp>
+#include <cstdint>
+#include "include/nlohmann/json.hpp"
+#include "matrix.hpp"
+
+using namespace boost::asio;
+
+// *********************************************************************
+// ||                             DEFINES                             ||
+// *********************************************************************
+#define POST_PORT 0xB00B
+
+// *********************************************************************
+// ||                            FUNCTIONS                            ||
+// *********************************************************************
+// AI did this, I have no idea if this is legit -Pat
+void handle_options_request(ip::tcp::socket& socket) {
+    // Send CORS headers for preflight request
+    std::string response = "HTTP/1.1 200 OK\r\n";
+    response += "Access-Control-Allow-Origin: *\r\n";
+    response += "Access-Control-Allow-Methods: DELETE, POST, GET, OPTIONS\r\n";
+    response += "Access-Control-Allow-Headers: Content-Type, Authorization, X-Requested-With\r\n";
+    response += "Content-Length: 0\r\n\r\n";
+    boost::asio::write(socket, boost::asio::buffer(response));
+}
+
```

Comment:
Ideally, once we have a prototype and stuff we can move all this code out of main into it's own api file/package akin to how I did it in Go